nav2_simple_commander/nav2_simple_commander/demo_picking.py

```diff
@@ -0,0 +1,123 @@
+# limitations under the License.
+
+import time
+from copy import deepcopy
+
+from geometry_msgs.msg import PoseStamped
+from rclpy.duration import Duration
+import rclpy
+
+from nav2_simple_commander.robot_navigator import BasicNavigator, NavigationResult
+
+# Shelf positions for picking
+shelf_positions = {
+    "shelf_A": [-3.829, -7.604],
+    "shelf_B": [-3.791, -3.287],
+    "shelf_C": [-3.791, 1.254],
+    "shelf_D": [-3.24, 5.861]}
+
+# Shipping destination for picked products
+shipping_destinations = {
+    "recycling": [-0.205, 7.403],
+    "pallet_jack7": [-0.073, -8.497],
+    "conveyer_432": [6.217, 2.153],
+    "frieght_bay_3": [-6.349, 9.147]}
+
+'''
+Basic item picking demo. In this demonstration, the expectation
+is that there is a person at the item shelf to put the item on the robot
+and at the pallet jack to remove it
+(probably with some kind of button for 'got item, robot go do next task').
+'''
+def main():
+    # Recieved virtual request for picking item at Shelf A and bring to
+    # worker at the pallet jack 7 for shipping. This request would
+    # contain the shelf ID ("shelf_A") and shipping destination ("pallet_jack7")
+    ####################
+    request_item_location = 'shelf_C'
+    request_destination = 'pallet_jack7'
+    ####################
+
+    rclpy.init()
+
+    navigator = BasicNavigator()
+
+    # Set our demo's initial pose
+    initial_pose = PoseStamped()
+    initial_pose.header.frame_id = 'map'
+    initial_pose.header.stamp = navigator.get_clock().now().to_msg()
+    initial_pose.pose.position.x = 3.45
+    initial_pose.pose.position.y = 2.15
+    initial_pose.pose.orientation.z = 1.0
+    initial_pose.pose.orientation.w = 0.0
+    navigator.setInitialPose(initial_pose)
+
+    # Wait for navigation to fully activate
+    navigator.waitUntilNav2Active()
+
+    shelf_item_pose = PoseStamped()
+    shelf_item_pose.header.frame_id = 'map'
+    shelf_item_pose.header.stamp = navigator.get_clock().now().to_msg()
+    shelf_item_pose.pose.position.x = shelf_positions[request_item_location][0]
+    shelf_item_pose.pose.position.y = shelf_positions[request_item_location][1]
+    shelf_item_pose.pose.orientation.z = 1.0
+    shelf_item_pose.pose.orientation.w = 0.0
+    print('Received request for item picking at ' + request_item_location + '.')
+    navigator.goToPose(shelf_item_pose)
+
+    # Do something during our route
+    # (e.x. queue up future tasks or detect person for fine-tuned positioning)
+    # Simply print information for workers on the robot's ETA for the demonstation
+    i = 0
+    while not navigator.isNavComplete():
+        i = i + 1
+        feedback = navigator.getFeedback()
+        if feedback and i % 5 == 0:
+            print('Estimated time of arrival at ' + request_item_location +
+                  ' for worker: ' + '{0:.0f}'.format(
+                  Duration.from_msg(feedback.estimated_time_remaining).nanoseconds / 1e9)
+                  + ' seconds.')
```

Comment:
Adding strings is harder to read than using arguments i.m.h.o.

```suggestion
            secs_left = Duration.from_msg(feedback.estimated_time_remaining).nanoseconds / 1e9)
            print(f'Estimated time of arrival at {request_item_location} for worker: {secs_left:.0f} seconds.')
```